main.go

```diff
@@ -18,40 +25,74 @@ import (
 )
 
 func main() {
+	log.SetFlags(log.Flags() | log.Lshortfile)
+	log.SetOutput(os.Stderr)
+
+	if err := run(); err != nil {
+		log.Fatalln("main failed to run:", err)
+	}
+
+	log.Println("sucessful shutdown")
+}
+
+func run() error {
 	config, err := LoadConfig()
 	if err != nil {
-		panic(err)
+		return fmt.Errorf("failed to LoadConfig: %w", err)
 	}
 
-	gin.SetMode(config.Env)
-
 	// migrate the db on startup
 	m, err := migrate.New("file://sql", config.DatabaseURL)
 	if err != nil {
-		panic(err)
+		return fmt.Errorf("failed to migrate.New: %w", err)
 	}
 
-	m.Up()
+	if err := m.Up(); err != nil {
+		if !errors.Is(err, migrate.ErrNoChange) {
+			return fmt.Errorf("failed to migrate Up: %w", err)
+		}
+
+		log.Println("no new migrations detected, schema is current")
+	}
 
 	db, err := sqlx.Open("postgres", config.DatabaseURL)
 	if err != nil {
-		panic(err)
+		return fmt.Errorf("failed to sqlx.Open: %w", err)
 	}
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	wg := sync.WaitGroup{}
+
+	wg.Add(1)
 	go func() {
+		defer wg.Done()
+		ticker := time.NewTicker(time.Hour)
```

Comment:
@sethetter, I wanted to point out that the behavior here has changed. Originally, the worker would run at startup, and then wait an hour to run again. Now, the worker will run after an hour has passed.

The reason I'm asking is because if the instance is being spun down before the hour, then this will likely never run, unless it's getting at least a drip of traffic to it. If that's the case, then we should go back to the previous behavior.

Reply:
Thanks for the call out -- I don't think there's much to worry about. There won't be cases where the service is _repeatedly_ run for less than a full hour and the cleanup never happening. If a job post stays up a few hours past when it was otherwise supposed to come down, there's no harm done.